reference/constraints/Bic.rst

```diff
@@ -0,0 +1,107 @@
++----------------+-----------------------------------------------------------------------+
+
+Basic Usage
+-----------
+
+To use the Bic validator, simply apply it to a property on an object that
+will contain a Business Identifier Code.
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Acme/SubscriptionBundle/Entity/Transaction.php
+        namespace Acme\SubscriptionBundle\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Transaction
+        {
+            /**
+             * @Assert\Bic(
+             *     message="This is not a valid Business Identifier Code (BIC)."
+             * )
+             */
+            protected $businessIdentifierCode;
+        }
+
+    .. code-block:: yaml
+
+        # src/Acme/SubscriptionBundle/Resources/config/validation.yml
+        Acme\SubscriptionBundle\Entity\Transaction:
+            properties:
+                businessIdentifierCode:
+                    - Bic:
+                        message: This is not a valid Business Identifier Code (BIC).
+
+    .. code-block:: xml
+
+        <!-- src/Acme/SubscriptionBundle/Resources/config/validation.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping http://symfony.com/schema/dic/constraint-mapping/constraint-mapping-1.0.xsd">
+
+            <class name="Acme\SubscriptionBundle\Entity\Transaction">
+                <property name="businessIdentifierCode">
+                    <constraint name="Bic">
+                        <option name="message">
+                            This is not a valid Business Identifier Code (BIC).
+                        </option>
+                    </constraint>
+                </property>
+            </class>
+        </constraint-mapping>
+
+    .. code-block:: php
+
+        // src/Acme/SubscriptionBundle/Entity/Transaction.php
+        namespace Acme\SubscriptionBundle\Entity;
+
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Transaction
+        {
+            protected $businessIdentifierCode;
+
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('businessIdentifierCode', new Assert\Bic(array(
+                    'message' => 'This is not a valid Business Identifier Code (BIC).',
+                )));
+            }
+        }
+
+Available Options
+-----------------
+
+message
+~~~~~~~
+
+**type**: ``string`` **default**: ``This is not a valid Business Identifier Code (BIC).``
+
+The default message supplied when the value does not pass the Bic check.
+
+.. include:: /reference/constraints/_payload-option.rst.inc
```

Comment:
"Upwards branch merging" is done from time to time in a non-formal schedule. Sometimes 2.8 and master are behind the other branchs, but at the end our doc managers always merge upwards.